llama_stack/providers/impls/meta_reference/datasetio/datasetio.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+from typing import List, Optional
+
+import pandas
+
+from llama_models.llama3.api.datatypes import *  # noqa: F403
+
+from llama_stack.apis.datasetio import *  # noqa: F403
+from dataclasses import dataclass
+
+from llama_stack.providers.datatypes import DatasetsProtocolPrivate
+from llama_stack.providers.utils.datasetio.dataset_utils import BaseDataset
+
+from .config import MetaReferenceDatasetIOConfig
+
+
+@dataclass
+class DatasetInfo:
+    dataset_def: DatasetDef
+    dataset_impl: BaseDataset
```

Comment:
what other kinds of datasets would be possible?

Reply:
For meta-reference, none that I can think of right now based on URL. For other providers (Huggingface/Pytorch), we  may need BaseDataset to implement the lightweight wrapper to get the rows.

Reply:
Moved `DatasetInfo` to `dataset_utils.py`